ntoskrnl/ke/i386/kiinit.c

```diff
@@ -392,6 +340,122 @@ KiInitializePcr(IN ULONG ProcessorNumber,
     Pcr->PrcbData.MultiThreadProcessorSet = Pcr->PrcbData.SetMember;
 }
 
+static
+CODE_SEG("INIT")
+VOID
+KiVerifyCpuFeatures(PKPRCB Prcb)
+{
+    CPU_INFO CpuInfo;
+
+    // 1. Check CPUID support
+    ULONG EFlags = __readeflags();
+
+    /* XOR out the ID bit and update EFlags */
+    ULONG NewEFlags = EFlags ^ EFLAGS_ID;
+    __writeeflags(NewEFlags);
+
+    /* Get them back and see if they were modified */
+    NewEFlags = __readeflags();
+
+    if (NewEFlags == EFlags)
+    {
+        /* The modification did not work, so CPUID is not supported. Try Windows 95. */
+        KeBugCheckEx(UNSUPPORTED_PROCESSOR, 0, 0, 0, 0); // TODO: what to put here?
```

Comment:
What to put: is the VendorString initialized at this stage? if so, something similar to what's done in the other bugchecks below?